### DISCUSSION TITLE
Presidio V2

### DISCUSSION THREAD

**Discussion by omri374** [microsoft/presidio#391] [General]
## Presidio V2 is coming!

In the upcoming weeks, we will start migrating the Presidio repo from the current version to a new version.

The main changes from V1 to V2 are:
- Replacing gRPC with HTTP to allow more customizable APIs and easier debugging.
- The anonymizer service will be Python based and pip installable.
- Focus on the analyzer and anonymizer service. Other services will be deprecated at first and potentially migrated over time to V2 with the help of the community.
- Better documentation, samples and build flows.

Notes:
- The current V1 code base will continue to be available but will no longer be officially supported.
- We will maintain backward compatibility with the current Presidio API for text analysis and anonymization.

We are certain this change will make Presidio much more accessible, maintainable and customizable and we look forward to collaborating with this great community!

Stay tuned!

The Presidio team

**Comment by rahulraina7** [comment]
@omri374  Does it make sense to use something like https://github.com/grpc-ecosystem/grpc-gateway
rather than completely getting rid of GRPC?

**Reply by omri374** [reply]
Thanks for the suggestion @rahulraina7. We'll look into this.

**Reply by rahulraina7** [reply]
Happy to contribute, if the plans are not formalized for this.

**Reply by imrj** [reply]
will there be a feature for discovery on traditional windows NTFS and Linux EXT file systems and the likes?

**Reply by omri374** [reply]
Hi @imrj, V2 services expose a Rest API to which you can send text input. Alternatively you can use Python file modules to read files and pass the input to Presidio. 
More generally, running and developing on both Windows and Linux is supported.
